bigtable-hbase-2.x-replication/src/main/java/com/google/cloud/bigtable/hbase2_x/replication/HbaseToCloudBigtableReplicationEndpoint.java

```diff
@@ -0,0 +1,67 @@
+package com.google.cloud.bigtable.hbase2_x.replication;
+
+import com.google.cloud.bigtable.hbase.replication.CloudBigtableReplicator;
+import com.google.cloud.bigtable.hbase.replication.adapters.BigtableWALEntry;
+import org.apache.hadoop.hbase.replication.BaseReplicationEndpoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.groupingBy;
+
+
+public class HbaseToCloudBigtableReplicationEndpoint extends BaseReplicationEndpoint {
```

Comment:
BaseReplicationEndpoint was marked as Private when they broke compatiblity on hbase 2.0 

" It undoes the LimitedPrivate on BaseReplicationEndpoint while keeping it for the RE Interface. It means consumers will have to copy/paste the AbstractService-based BRE into their own codebase also supplying their  own Guava;"
https://issues.apache.org/jira/browse/HBASE-15982

You should implement the ReplicationEndpoint interface and implement/copy-paste the guava service as they suggested in the jira.


BaseReplicationEndpoint is `Private`
https://github.com/apache/hbase/blob/9f7e856a5f84a6f52d0197a03b7a56e1a3f0446c/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/BaseReplicationEndpoint.java#L36

ReplicationEndpoint is `LimitedPrivate` and we should use it.
https://github.com/apache/hbase/blob/9f7e856a5f84a6f52d0197a03b7a56e1a3f0446c/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/ReplicationEndpoint.java#L50

Reply:
upcoming PR.